src/classes/CollisionSolution.ts

```diff
@@ -0,0 +1,203 @@
+import { Battlefield } from './Battlefield';
+import { Bullet } from './Bullet';
+import { CONFIG } from './Config';
+import { Tank } from './Tank';
+
+export class CollisionSolution {
+  public wallList: number[];
+  public tankList: Tank[];
+  public bulletList: Bullet[];
+  public battlefield: Battlefield;
+
+  public constructor(battlefield: Battlefield) {
+    this.wallList = [0, 0, 0, 0];
+    this.tankList = [];
+    this.bulletList = [];
+    this.battlefield = battlefield;
+  }
+
+  public updateBattlefield(battlefield: Battlefield): void {
+    this.battlefield = battlefield;
+    const wall: number[] = [];
```

Comment:
I see that you initialize array but it never used

Reply:
deleted. this method was planned to test the reduction of the zone